File: integration-tests/java-selenium/java-selenium-test/src/main/java/com/gu/fronts/integration/test/page/common/AllFaciaContainersPage.java
package com.gu.fronts.integration.test.page.common;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute;
import com.gu.fronts.integration.test.page.util.AbstractParentPage;

public class AllFaciaContainersPage extends AbstractParentPage {
    @FindByTestAttribute(using = "all-front-containers")
    private WebElement allFaciaContainers;

    public AllFaciaContainersPage(WebDriver webDriver) {
        super(webDriver);
    }

    @Override
    public AllFaciaContainersPage isDisplayed() {
        assertExistsAndDisplayed(allFaciaContainers);
        return this;
    }

    public FaciaContainer containerWithId(String testAttributeId) {
        return pageFactory.initPage(webDriver, FaciaContainer.class, allFaciaContainers, testAttributeId);
    }
}

File: integration-tests/scala-selenium/app/com/gu/fronts/integration/test/page/common/AllFaciaContainersPage.scala
package com.gu.fronts.integration.test.page.common

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute
import com.gu.fronts.integration.test.page.util.AbstractParentPage
import com.gu.fronts.integration.test.page.util.FrontsParentPage
import com.gu.fronts.integration.test.page.util.PageElementHelper.existsAndDisplayed

class AllFaciaContainersPage(webDriver: WebDriver) extends FrontsParentPage(webDriver) {

  @FindByTestAttribute(using = "all-front-containers")
  private var allFaciaContainers: WebElement = _

  override def isDisplayed(): Boolean = {
    existsAndDisplayed(allFaciaContainers)
  }

  def containerWithId(testAttributeId: String): FaciaContainer = {
    pageFactory.initPage(webDriver, classOf[FaciaContainer], allFaciaContainers, testAttributeId)
  }
}